tests/integration/iface_admin_state_test.py

```diff
@@ -0,0 +1,38 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+import copy
+
+from libnmstate import netapplier
+
+from .testlib import assertlib
+from .testlib.statelib import INTERFACES
+
+
+def test_set_a_down_iface_down():
+    desired_state = {
+        INTERFACES: [
+            {
+                'name': 'eth1',
+                'type': 'ethernet',
+                'state': 'down',
+            }
+        ]
+    }
+    netapplier.apply(copy.deepcopy(desired_state))
```

Comment:
How about applying the state twice to ensure that the interface is really down when it is set down again.

Reply:
The fixture assures that: that is the base state.
But I can add an assert here to make sure we do start from the correct state in the test.

Reply:
The assert would be great to make this obvious.

Reply:
See #122

Reply:
I added it